### ISSUE TITLE
Database Backend Should Support the retry config params`

### ISSUE THREAD

**Issue by vinayan3** [celery/celery#6540]
# Checklist

- [X] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical enhancement to an existing feature.
- [X] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed enhancements.
- [X] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the if the same enhancement was already implemented in the
  master branch.
- [X] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Brief Summary
Database backend allow the base class retries to kick introduced by https://github.com/celery/celery/pull/6122 . It introduced the following params:
* `result_backend_always_retry` 
* `result_backend_max_sleep_between_retries_ms`
* `result_backend_base_sleep_between_retries_ms` 
* `result_backend_max_retries` 

The Database backend has it's own retries 
https://github.com/celery/celery/blob/8bceb446e6a07682d4b8dd6199cdac450bd63578/celery/backends/database/__init__.py#L39-L60

There is currently some discrepancies between the BaseBackend retries and the Database backend retries. Specifically the methods for save_group, restore_group, delete_group, and forget have retries in the Database backend while the BaseBackend does not.

# Design
The design work has already been done in PR https://github.com/celery/celery/pull/6122. 

## Architectural Considerations
Do any other backends need to be touched up to fully support the config?

## Proposed Behavior
Used the configuration for the how the backend should retry in the database backend. This will require implementing the method `exception_safe_to_retry`.

## Proposed UI/UX
N/A 

## Diagrams
N/A
## Alternatives
None

**Comment by auvipy** [comment]
are you interested to start the effort?

**Comment by ChickenBenny** [comment]
Hey @auvipy @vinayan3, I've put together a PR #10130 to fix this issue.

**Comment by auvipy** [comment]
did you address the following? Used the configuration for the how the backend should retry in the database backend. This will require implementing the method exception_safe_to_retry.

**Comment by ChickenBenny** [comment]
Yes! The `exception_safe_to_retry` method is actually already implemented in `DatabaseBackend` (it currently checks `isinstance(exc, RETRYABLE_DB_ERRORS)`). We can find this in [here](https://github.com/celery/celery/blob/main/celery/backends/database/__init__.py#L132).

Since my PR refactors the backend to use `BaseBackend._ensure_retryable`, the unified retry loop will correctly invoke this existing `exception_safe_to_retry` method to properly identify and retry recoverable database errors.

However, because `BaseBackend` defaults to `always_retry = False`, existing DatabaseBackend users would lose their implicit automatic retries unless they update their configs.

To prevent this breaking change, I plan to override these defaults in `DatabaseBackend.__init__`. Setting `always_retry = conf.get('result_backend_always_retry', True)` and `max_retries = conf.get('result_backend_max_retries', 3)` will preserve the previous behavior, ensuring zero disruption for current users.

What do you think about this approach?
